### PR TITLE
Refactoring DraftTransaction to TransactionNext

### PR DESCRIPTION
**Pull request by ramonalobont**

### CODE REVIEW

**Review by andreibancioiu — COMMENTED** [review]
Now I see, we should have not removed the transaction builder. Sorry for incorrectly describing the steps. They should have been:

 - adjust (add chain ID) and rename the transaction builder with respect to: https://github.com/multiversx/mx-sdk-py-core/blob/main/multiversx_sdk_core/transaction_factories/transaction_builder.py
 - use `TransactionNext` instead `DraftTransaction`
 - remove `DraftTransaction`
 - rename remaining variables (e.g. to not contain "draft" anymore).

That is, no other code movement should happen - the gas computation logic and so on should stay as they were.

For reference: https://github.com/multiversx/mx-sdk-py-core/tree/main/multiversx_sdk_core/transaction_factories

File: src/transactionsFactories/transferTransactionsFactory.ts [hunk 1]
```diff
@@ -69,13 +71,11 @@ export class NextTransferTransactionsFactory {
         const transferArgs = this.dataArgsBuilder.buildArgsForMultiESDTNFTTransfer(
             options.receiver,
             options.tokenTransfers
-        );
+        )
```
Comment (popenta) [inline src/transactionsFactories/transferTransactionsFactory.ts:74]: add `;` after calling the method.

File: src/transactionsFactories/tokenManagementTransactionsFactory.ts [hunk 2]
```diff
@@ -603,4 +623,4 @@ IMPORTANT!
 You are about to issue (register) a new token. This will set the role "ESDTRoleBurnForAll" (globally).
 Once the token is registered, you can unset this role by calling "unsetBurnRoleGlobally" (in a separate transaction).`);
     }
-}
+}
```
Comment (popenta) [inline src/transactionsFactories/tokenManagementTransactionsFactory.ts:626]: add empty line at end of file

File: src/transactionsFactories/tokenManagementTransactionsFactory.ts [hunk 3]
```diff
@@ -120,7 +122,8 @@ export class TokenManagementTransactionsFactory {
             options.canAddSpecialRoles ? this.trueAsHex : this.falseAsHex,
         ];
 
-        return new DraftTransactionBuilder({
+
```
Comment (popenta) [inline src/transactionsFactories/tokenManagementTransactionsFactory.ts:125]: Remove one empty line unless the formatter added it, then you can ignore the comments. Applies to all methods.

File: src/transactionsFactories/delegationTransactionsFactory.spec.ts [hunk 4]
```diff
@@ -10,28 +10,28 @@ describe("test delegation transactions factory", function () {
...
 
-    it("should create draft transaction for new delegation contract", async function () {
+    it("should create 'TransactionNext' for new delegation contract", async function () {
         const sender = Address.fromBech32("erd18s6a06ktr2v6fgxv4ffhauxvptssnaqlds45qgsrucemlwc8rawq553rt2");
         const delagationCap = "5000000000000000000000";
         const serviceFee = 10;
         const value = new BigNumber("1250000000000000000000");
 
-        const draft = delegationFactory.createTransactionForNewDelegationContract({
+        const next = delegationFactory.createTransactionForNewDelegationContract({
```
Comment (popenta) [inline src/transactionsFactories/delegationTransactionsFactory.spec.ts:19]: Could be renamed to `transaction`. (optional) Same for the other tests.

File: src/transactionsFactories/delegationTransactionsFactory.ts [hunk 5]
```diff
@@ -19,33 +18,27 @@ interface Config {
     additionalGasLimitPerValidatorNode: BigNumber.Value;
     additionalGasLimitForDelegationOperations: BigNumber.Value;
 }
-
```
Comment (popenta) [inline src/transactionsFactories/delegationTransactionsFactory.ts:22]: unless the formatter removed the empty lines they can be added back.

File: src/transactionsFactories/delegationTransactionsFactory.ts [hunk 6]
```diff
@@ -349,9 +304,8 @@ export class DelegationTransactionsFactory {
             addDataMovementGas: true
         }).build();
     }
-
     private computeExecutionGasLimitForNodesManagement(numNodes: number): BigNumber.Value {
         const additionalGasForAllNodes = new BigNumber(this.config.additionalGasLimitPerValidatorNode).multipliedBy(numNodes);
         return new BigNumber(this.config.gasLimitDelegationOperations).plus(additionalGasForAllNodes);
     }
-}
+}
```
Comment (popenta) [inline src/transactionsFactories/delegationTransactionsFactory.ts:311]: add empty line

File: src/transactionsFactories/smartContractTransactionsFactory.ts [hunk 7]
```diff
@@ -185,4 +183,4 @@ export class SmartContractTransactionsFactory {
         }
         return true;
     }
-}
+}
```
Comment (popenta) [inline src/transactionsFactories/smartContractTransactionsFactory.ts:186]: add empty new line

**Review by andreibancioiu — APPROVED** [review]

**Review by popenta — APPROVED** [review]

**Pull request merged**